### PR TITLE
rbd-mirror: removing non-primary trash snapshot

### PR DESCRIPTION
**Pull request by trociny**
## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`

</details>

### CODE REVIEW

**Comment by trociny** [comment]
rebased after #31161 is merged.

File: src/librbd/ExclusiveLock.h [hunk 1]
```diff
@@ -18,7 +19,9 @@ class ExclusiveLock : public ManagedLock<ImageCtxT> {
 
   ExclusiveLock(ImageCtxT &image_ctx);
 
-  bool accept_requests(int *ret_val = nullptr) const;
+  bool accept_requests() const;
```
Comment (dillaman) [inline src/librbd/ExclusiveLock.h]: Nit: can we drop this and just update `librbd::Operations` to call the new method w/ the correct enum?

File: src/librbd/ExclusiveLock.cc [hunk 2]
```diff
@@ -37,20 +37,35 @@ ExclusiveLock<I>::ExclusiveLock(I &image_ctx)
...
+bool ExclusiveLock<I>::accept_request(OperationRequestType request_type,
+                                      int *ret_val) const {
+  std::lock_guard locker{ML<I>::m_lock};
+
+  auto policy = m_image_ctx.get_exclusive_lock_policy();
+  bool accept_request = (!ML<I>::is_state_shutdown() &&
+                         ML<I>::is_state_locked() &&
+                         (m_request_blocked_count == 0 ||
+                          policy->accept_blocked_request(request_type)));
+  *ret_val = m_request_blocked_ret_val;
```
Comment (dillaman) [inline src/librbd/ExclusiveLock.cc]: Nit: (for clarity) probably don't want to return an error code if accepts are being accepted.

**Comment by dillaman** [comment]
jenkins test make check

**Comment by trociny** [comment]
@dillaman updated

**Comment by dillaman** [comment]
jenkins test make check

**Comment by dillaman** [comment]
jenkins retest make check

**Comment by dillaman** [comment]
jenkins test make check

**Pull request merged**